index.js

```diff
@@ -155,9 +155,18 @@ Notifications.localNotification = function(details: Object) {
  */
 Notifications.localNotificationSchedule = function(details: Object) {
 	if ( Platform.OS === 'ios' ) {
+
+		let soundName = details.soundName ? details.soundName : 'default'; // play sound (and vibrate) as default behaviour
+
+		if (details.hasOwnProperty('playSound') && !details.playSound) {
+			soundName = ''; // empty string results in no sound (and no vibration)
+		}
+
 		this.handler.scheduleLocalNotification({
 			fireDate: details.date.toISOString(),
 			alertBody: details.message,
+			soundName: soundName,
+			applicationIconBadgeNumber: parseInt(details.number, 10),
```

Comment:
To indicate that the string input number to parse is of base 10.

10 is default anyway, but people like to be explicit.